lib/Vector3/Vector3.cpp

```diff
@@ -100,3 +100,10 @@ Vector3 Vector3::normalized() {
 		return Vector3();
 	}
 }
+
+void Vector3::randomize(std::default_random_engine& generator,
```

Comment:
What is this function for, randomize? Just testing?

Reply:
Yep, it randomizes the three scalars in the `Vector3` instance using random numbers generated by `std::default_random_engine`. 

Every `Sensor` has its own instance of the  `std::default_random_engine` object created with a standard deviation and mean output value (created when the `Sensor` object is initialized). The random engine for the particular `Sensor` gets piped into this function (as a reference because of `&` symbol) to be used in randomizing the vector.